### PR TITLE
Create a copy of the Bytes passed to Resource::new_from_data() if it'…

### PR DESCRIPTION
**Pull request by sdroege**
…s not pointer-aligned

See https://bugzilla.gnome.org/show_bug.cgi?id=790030

Fixes https://github.com/gtk-rs/glib/issues/242

### CODE REVIEW

File: src/resource.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            // Create a copy of data if it is not pointer-aligned
+            // https://bugzilla.gnome.org/show_bug.cgi?id=790030
+            let mut data = data.clone();
+            let data_ptr = glib_ffi::g_bytes_get_data(data.to_glib_none().0, ptr::null_mut());
+            if data_ptr as usize % mem::align_of::<*const u8>() != 0 {
+                data = glib::Bytes::from(&*data);
+            }
+
+            let ret = ffi::g_resource_new_from_data(data.to_glib_none().0, &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
```
Comment (GuillaumeGomez) [inline src/resource.rs]: No one liner please. :)

**Comment by GuillaumeGomez** [comment]
I followed the discussion on IRC and this seems to do what's expected. Just waiting for @EPashkin's confirmation.

File: src/resource.rs (hunk 1 above)
Comment (sdroege) [reply src/resource.rs]: That's how it came from the autogenerated code (these kind of review comments that are wasting both our time are why I believe we should just use rustfmt and get over it...)

Comment (GuillaumeGomez) [reply src/resource.rs]: Thanks. And I told you that I wasn't completely opposed to rustfmt. If you have the patience and the time to set one up, then we'll use it.

**Comment by EPashkin** [comment]
@sdroege Thanks.

@GuillaumeGomez  👍 from me

**Pull request merged**